File: ResourceManager/Models/Configurations/BookTagConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class BookTagConfiguration : IEntityTypeConfiguration<BookTag>
    {
        public void Configure(EntityTypeBuilder<BookTag> builder)
        {
            builder.ToTable("BookTags");
            builder.HasKey(x => new { x.BookID, x.TagID });

            builder.HasOne(x => x.Book)
                .WithMany(x => x.BookTags)
                .HasForeignKey(x => x.BookID);

            builder.HasOne(x => x.Tag)
                .WithMany(x => x.BookTags)
                .HasForeignKey(x => x.TagID);

        }
    }
}

File: ResourceManager/Models/Configurations/BookTagConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore

Namespace ResourceManager.Models.Configurations
    Public Class BookTagConfiguration
        Implements IEntityTypeConfiguration(Of BookTag)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of BookTag)) Implements IEntityTypeConfiguration(Of BookTag).Configure
            builder.ToTable("BookTags")
            builder.HasKey(Function(x) New With {x.BookID, x.TagID
            })

            builder.HasOne(Function(x) x.Book).WithMany(Function(x) x.BookTags).HasForeignKey(Function(x) x.BookID)

            builder.HasOne(Function(x) x.Tag).WithMany(Function(x) x.BookTags).HasForeignKey(Function(x) x.TagID)

        End Sub
    End Class
End Namespace
